lib/wire-protocol.d.ts

```diff
@@ -134,74 +154,41 @@ export declare class QueryTimeoutError extends ServiceError {
   readonly stats?: {
     [key: string]: number;
   };
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: 440;
-    summary?: string;
-    stats?: {
-      [key: string]: number;
-    };
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
 }
 /**
  * AuthenticationError indicates invalid credentials were
  * used.
  */
 export declare class AuthenticationError extends ServiceError {
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: 401;
-    summary?: string;
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
 }
 /**
  * AuthorizationError indicates the credentials used do not have
  * permission to perform the requested action.
  */
 export declare class AuthorizationError extends ServiceError {
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: 403;
-    summary?: string;
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
 }
 /**
  * ThrottlingError indicates some capacity limit was exceeded
  * and thus the request could not be served.
  */
 export declare class ThrottlingError extends ServiceError {
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: 429;
-    summary?: string;
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
 }
 /**
  * ServiceInternalError indicates Fauna failed unexpectedly.
  */
 export declare class ServiceInternalError extends ServiceError {
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: 500;
-    summary?: string;
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
 }
 /**
  * ServiceTimeoutError indicates Fauna was not available to servce
  * the request before the timeout was reached.
  */
 export declare class ServiceTimeoutError extends ServiceError {
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: 503;
-    summary?: string;
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
```

Comment:
we should enforce that `httpStatus` can only take the correct status code in these consturctors.

So replace `number` with `503`; same for all the other errors.

Reply:
got it. I was too aggressive with the copy-pasta. I'll fix that up.